internal/kafka/command_topic_produce.go

```diff
@@ -231,6 +232,31 @@ func serializeMessage(keyMetaInfo, valueMetaInfo []byte, data, delimiter string,
 	return append(keyMetaInfo, serializedKey...), append(valueMetaInfo, serializedValue...), nil
 }
 
+func getKeyAndValue(schemaBased bool, data, delimiter string) (string, string, error) {
+	dataSplit := strings.Split(data, delimiter)
+	if len(dataSplit) < 2 {
+		return "", "", errors.New(errors.MissingKeyOrValueErrorMsg)
+	}
+
+	if !schemaBased {
+		return strings.TrimSpace(dataSplit[0]), strings.TrimSpace(strings.Join(dataSplit[1:], delimiter)), nil
+	}
+
+	key := dataSplit[0]
+	if json.Valid([]byte(strings.TrimSpace(key))) {
+		return strings.TrimSpace(key), strings.TrimSpace(strings.Join(dataSplit[1:], delimiter)), nil
+	}
+
+	for i, substr := range dataSplit[1:] {
+		key += delimiter + substr
```

Comment:
@brianstrauch I went with this instead of the bracket matching idea for now because that requires more care: bracket characters appearing inside quotations need to be ignored, which means quotation marks need to be matched, which itself means we need to be careful about escaped quotation marks and backslashes, etc.

Reply:
Cool, I guess this is O(n)? Hopefully won't take too long for really big inputs

Reply:
Yeah I'm hoping that it's not too bad on average, since I don't expect users to be entering really complex keys. And I don't validate the formatting of the value/message, which I let the serializer do.